tests/test_model_specs.py

```diff
@@ -57,35 +57,184 @@ def test_model_specs_are_valid(self):
+                        spec['fields'][field],
+                        f'{key}.fields.{field}',
+                        parent_type=t)
+
+                self.assertIn('geometries', spec)
+                self.assertIsInstance(spec['geometries'], set)
+
+                attrs.remove('fields')
+                attrs.remove('geometries')
+
+            elif t == 'csv':
+                # csv type may have a columns property.
+                # the columns property maps each expected column header
+                # name/pattern to a nested type dictionary describing the data
+                # in that column. may be absent if the table structure
+                # is too complex to describe this way.
+                self.assertIn('columns', spec)
+                self.assertIsInstance(spec['columns'], dict)
+                for column in spec['columns']:
+                    self.assertIsInstance(column, str)
+                    self.validate_output(
+                        spec['columns'][column],
+                        f'{key}.columns.{column}',
+                        parent_type=t)
+                attrs.discard('columns')
+
+            elif t == 'directory':
+                # directory type should have a contents property that maps each
+                # expected path name/pattern within the directory to a nested
+                # type dictionary describing the data at that filepath
+                self.assertIn('contents', spec)
+                self.assertIsInstance(spec['contents'], dict)
+                for path in spec['contents']:
+                    self.assertIsInstance(path, str)
+                    self.validate_output(
+                        spec['contents'][path],
+                        f'{key}.contents.{path}',
+                        parent_type=t)
+                attrs.remove('contents')
+
+            elif t == 'option_string':
+                    # option_string type should have an options property that
+                    # describes the valid options
+                    self.assertIn('options', spec)
+                    self.assertIsInstance(spec['options'], dict)
+                    for option, description in spec['options'].items():
+                        self.assertTrue(
+                            isinstance(option, str) or
+                            isinstance(option, int))
+                        self.assertIsInstance(description, str)
+                    attrs.remove('options')
+
+            elif t == 'file':
+                pass
+
+            # iterate over the remaining attributes
+            # type-specific ones have been removed by this point
+            if 'about' in attrs:
+                self.assertIsInstance(spec['about'], str)
+                attrs.remove('about')
+            if 'created_if' in attrs:
+                # required value may be True, False, or a string that can be
+                # parsed as a python statement that evaluates to True or False
```

Comment:
Maybe, just a value that can be evaluated to a python boolean?

Reply:
Tried to rephrase this better